intrastat_product/models/account_move.py

```diff
@@ -90,17 +90,61 @@ def _get_intrastat_line_vals(self, line):
             if not hs_code:
                 return vals
             weight, qty = decl_model._get_weight_and_supplunits(line, hs_code)
+            if not self.env.context.get("bypass_transaction_weight_convert"):
```

Comment:
Why this?

Reply:
This is necessary to prevent weight being converted to an integer by default.

Reply:
Yes, but why convert it to int? Let the decimal precision to act